app/views/issues/_toolbar.html.erb

```diff
@@ -0,0 +1,41 @@
+<%#
+  Inspired in the Gmail toolbar. Some of the controls are hidden while nothing
+  is selected and they become visible once the user selects at least one row.
+%>
+<div class="btn-toolbar">
+  <div class="btn-group">
+     <button class="btn">
+       <input type="checkbox" id="select-all" />
+     </button>
+  </div>
+
+  <div class="btn-group issue-actions js-issue-actions">
+    <a id="delete-selected" class="btn" data-confirm="Are you sure?" data-method="delete"><i class="fa fa-trash text-error"></i></a>
+    <a class="btn dropdown-toggle" data-toggle="dropdown">
+      <i class="fa fa-tags"> </i>
+      <span class="caret"></span>
+    </a>
+    <ul class="dropdown-menu">
+      <% @tags.each do |tag| %>
+      <li>
+        <%= link_to 'javascript:void(0)', class: 'js-taglink', style: "color: #{tag.color}", data: { tag: h(tag.name) } do %>
+          <i class="fa fa-tag"></i> <%= h(tag.display_name) %>
+        <% end %>
+      </li>
+      <% end %>
+    </ul>
+  </div>
+
+  <div class="btn-group">
+    <button class="btn" data-toggle="dropdown"><i class="fa fa-columns"></i> <span class="caret"></span></button>
+    <ul class="dropdown-menu js-table-columns">
+      <li><a href="javascript:void(0)" data-column="title"><input type="checkbox" /> Title</a></li>
```

Comment:
Bug:

1. Click the dropdown and hide some columns
2. Click 'select all'
3. Click the dropdown again. All 3 columns are checked, but the ones you unchecked in step 1 will still be hidden in the table. When you "uncheck" a checkbox for a column that's actually hidden, the column will appear again but the checkbox will stay clicked.

The same bug also happens in reverse: if you select all issues, hide some columns, then uncheck the 'select all' checkbox, then all columns will be unchecked in the dropdown even if some the actual columns in the table are still visible